controllers/shipwrightbuild_controller.go

```diff
@@ -104,6 +104,44 @@ func deleteObjectsIfPresent(ctx context.Context, k8sClient client.Client, objs [
 	return nil
 }
 
+// Check and fetch TektonConfig
+func (r *ShipwrightBuildReconciler) fetchAndCheckTektonConfig(ctx context.Context, logger logr.Logger, b *v1alpha1.ShipwrightBuild) (bool, error) {
+	tektonConfig, err := r.TektonOperatorClient.TektonConfigs().Get(ctx, "config", metav1.GetOptions{})
+	if err != nil {
```

Comment:
Please differentiate between the "not found" error and other types of errors:

```suggestion
	tektonConfig, err := r.TektonOperatorClient.TektonConfigs().Get(ctx, "config", metav1.GetOptions{})
	if kerrors.IsNotFound(err) {
```

We can get other types of errors trying to get data from the kube apiserver (including timeouts). These we typically do not record to status because they are transient. TektonConfig missing is something that is more permanent and might require intervention from a cluster admin.